packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts

```diff
@@ -120,6 +121,8 @@ export abstract class DeployExecutor<T> extends DeployRetrieveExecutor<T> {
     components: ComponentSet,
     token: vscode.CancellationToken
   ): Promise<DeployResult | undefined> {
+    Lifecycle.getInstance().removeAllListeners('scopedPostDeploy');
+    Lifecycle.getInstance().removeAllListeners('scopedPreDeploy');
```

Comment:
I'm pretty sure that removing these listeners will have a negative effect on subsequent calls to "SFDX: View Local Changes" and the other "View Changes" commands.  STL will get out of sync and report an incorrect status.  I am going to verify and also explore another option, managing a single instance of STL, which is being explored in [this PR](https://github.com/forcedotcom/salesforcedx-vscode/pull/5115).